clients/lsp-roslyn.el

```diff
@@ -0,0 +1,363 @@
+;;; lsp-roslyn.el --- description -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2023 Ruin0x11
+
+;; Author: Ruin0x11 <ipickering2@gmail.com>
+;; Keywords:
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; C# client using the Roslyn language server
+
+;;; Code:
+
+(require 'lsp-mode)
+(require 'cl-lib)
+(require 'f)
```

Comment:
We probably don't need to require f and cl-lib here since lsp-mode already has those as dependencies.